tests/sentry/grouping/ingest/test_seer.py

```diff
@@ -157,12 +165,40 @@ def test_obeys_customized_fingerprint_check(self) -> None:
             (custom_fingerprint_event, False),
             (built_in_fingerprint_event, False),
         ]:
-
             assert (
                 should_call_seer_for_grouping(event, event.get_grouping_variants())
                 is expected_result
             ), f'Case with fingerprint {event.data["fingerprint"]} failed.'
 
+    @patch("sentry.grouping.ingest.seer.metrics")
+    def test_obeys_empty_stacktrace_string_check(self, mock_metrics: Mock) -> None:
+        self.project.update_option("sentry:similarity_backfill_completed", int(time()))
+        new_event = Event(
+            project_id=self.project.id,
+            event_id="12312012112120120908201304152013",
+            data={
+                "title": "title",
+                "platform": "python",
+                "stacktrace": {"in-app": False, "contributes": False, "frames": [{}]},
+            },
+        )
+
+        assert should_call_seer_for_grouping(new_event, new_event.get_grouping_variants()) is False
+        sample_rate = options.get("seer.similarity.metrics_sample_rate")
+        mock_metrics.incr.assert_any_call(
+            "grouping.similarity.did_call_seer",
+            sample_rate=sample_rate,
+            tags={
+                "call_made": False,
+                "blocker": "empty-stacktrace-string",
+            },
+        )
+
+    def test_stacktrace_string_not_saved_in_event(self) -> None:
+        self.project.update_option("sentry:similarity_backfill_completed", int(time()))
+        should_call_seer_for_grouping(self.event, self.event.get_grouping_variants())
+        assert self.event.data.get("stackrace") is None
```

Comment:
I think to be realistic, we should probably actually save the event. What do you think of something like this?

```suggestion
    @patch("sentry.grouping.ingest.seer.get_similarity_data_from_seer", return_value=[])
    def test_stacktrace_string_not_saved_in_event(
        self, mock_get_similarity_data: MagicMock
    ) -> None:
        self.project.update_option("sentry:similarity_backfill_completed", 11212012)
        event = save_new_event(self.event_data, self.project)

        # We sent the string
        assert mock_get_similarity_data.call_count == 1
        assert "raise FailedToFetchError('Charlie didn't bring the ball back')" in (
            mock_get_similarity_data.call_args.args[0]["stacktrace"]
        )
        # But it didn't end up in the event
        assert event.data.get("stacktrace_string") is None
```